### PR TITLE
Nested ambient transaction

### PR DESCRIPTION
**Pull request by dshuvaev**
Allowed several nested ambient transactions to be processed. Main scenario is when one transaction with TransactionScopeOption.RequiresNew is created inside another one. Thus when child transaction is complete its results are persisted, while parent transaction can be rejected.

### CODE REVIEW

**Comment by ajcvickers** [comment]
@dshuvaev You need to sign the CLA before we can look at this PR.

**Comment by dshuvaev** [comment]
> 
> 
> @dshuvaev You need to sign the CLA before we can look at this PR.

I am getting approval to do so.

**Comment by dshuvaev** [comment]
> 
> 
> @dshuvaev You need to sign the CLA before we can look at this PR.

@ajcvickers It is signed. Thank you

File: src/EFCore.Relational/Storage/RelationalConnection.cs [hunk 1]
```diff
@@ -548,11 +553,16 @@ private void ClearTransactions(bool clearAmbient)
         {
             CurrentTransaction = null;
             EnlistedTransaction = null;
-            if (clearAmbient
-                && _ambientTransaction != null)
+            if (clearAmbient && _ambientTransactions.Any())
```
Comment (smitpatel) [inline src/EFCore.Relational/Storage/RelationalConnection.cs]: `_ambientTransactions.Count > 0`

**Comment by AndriySvyryd** [comment]
@dshuvaev Thanks for your contribution!

File: src/EFCore.Relational/Storage/RelationalConnection.cs [hunk 2]
```diff
@@ -629,53 +639,52 @@ var interceptionResult
         private void HandleAmbientTransactions()
         {
             var current = Transaction.Current;
-            if (current != null
-                && !SupportsAmbientTransactions)
+
+            if (current == null) 
             {
-                Dependencies.TransactionLogger.AmbientTransactionWarning(this, DateTimeOffset.UtcNow);
+                var rootTransaction = _ambientTransactions.Any() && _ambientTransactions.TryPeek(out var transaction) ? transaction : null;
```
Comment (smitpatel) [inline src/EFCore.Relational/Storage/RelationalConnection.cs]: could make count check.

**Pull request merged**